storage/innobase/srv/srv0srv.cc

```diff
@@ -2870,7 +2870,18 @@ bool srv_enable_redo_encryption_rk(THD *thd) {
   space->encryption_redo_key = mkey;
   space->flags |= FSP_FLAGS_MASK_ENCRYPTION;
   space->encryption_key_version = version;
-  memcpy(space->encryption_redo_key_uuid, server_uuid,
+  space->encryption_redo_key_uuid.reset(
+      new (std::nothrow) char[ENCRYPTION_SERVER_UUID_LEN + 1]);
+  if (space->encryption_redo_key_uuid.get() == nullptr) {
```

Comment:
btw, should you do reset() on fil_space_t destruction.. Although this is done on shutdown only.. for redo log tablespaces..

Reply:
that's the point of smart_pointer  - when it gets destructed - it will deallocate the memory it points to.

Reply:
and it will get destructed when fil_space_t gets destructed.

Reply:
Right.. reset() is only required f you want to deallocate explicity..